[FIX] l10n_ch,web,base: improve QR-BILL report matching standard

nle-odoo opened the pull request:
Improve placement of items according to:

 "Swiss Implementation Guidelines QR-bill"

In particular:

- "Payment part" amount under QR barcode instead of under payment part
  information section

- QR Code that does 46mm x 46mm and 56mm x 56mm minus margin (that can
  be variable depending on QR version/content).

- removing procedure zone with "Supports: Credit transfer"

- "amount" and "acceptance point" of receipt at right location.

 This is only for version below 13.0, in 13.0 all these are fixed with
 a3029a8c9b9.

 Depending on how the instance is updated:

 - without "-u l10n_ch" => "amount" and "acceptance point" will stay at
   their location and not right location

- without restarting odoo => QR barcode will have correct size minus
  variable margin (so 46mm x 46mm minus variable margin instead of 56mm
  x 56mm minus variable margin before change)

To make barcode work without margin (for the fixed size of 46mm), this
PR adds "QR_quiet" type that is "QR" but with no margin: this is only
temporary for 12.0 and saas-12.3, as of 13.0 there is a "quiet" method
argument that do the exact same thing.

opw-2288880

nle-odoo commented:
## before change

![](https://user-images.githubusercontent.com/9977887/87412005-c48e1b80-c5c8-11ea-91ab-cb8ee3a17728.jpg)

## after change

![](https://user-images.githubusercontent.com/9977887/87411509-1b472580-c5c8-11ea-9c79-cf699be51608.png)

## standard

![](https://user-images.githubusercontent.com/9977887/87410694-edadac80-c5c6-11ea-8cf3-37835db5b42d.png)

nle-odoo commented:
fw-bot up to saas-12.3

(this is already solved in 13.0 with https://github.com/odoo/odoo/commit/a3029a8c9b9816382f217741cef311b97eecdd52)

yvaucher commented:
@nle-odoo Looks much better :1st_place_medal: 

:warning: This might not be the aim of this PR but from what I can see visually, the content of the **Reference** should only include a structured Ref. (QRR type in our case as the SCOR type is not implemented)

Any non structured reference must appear in **Additional information**


I also see that the amount on the left slip is not displayed as on the payment part. Wheras in the specs it's 2 columns, with an explicit "Currency" label.


Please also update the translations, you have all translations available in the official documentation.

As a reminder, we want it translated in FR, DE, IT

You might want to take directly the terms from my PR in version 13.0: https://github.com/odoo/odoo/pull/53084


:information_source: Additional note about the QR code. The QR version used here might be a bit too small in case you want to add the swico standard, that adds payment terms and taxes informations in the QR Code data.

nle-odoo commented:
> I just passed the QR bill screen capture you have in a validator and it's badly formatted, Is it only for tests purposes?

I patched the methods and view to print even with wrong data for test purposes, the QR code should contain this data:

```
yeah1=yeah2
yeah3=yeah4
```

> I also see that the amount on the left slip is not displayed as on the payment part. Wheras in the specs it's 2 columns, with an explicit "Currency" label.

I added it on 2 columns in receipt also

> Please also update the translations, you have all translations available in the official documentation.
> As a reminder, we want it translated in FR, DE, IT

I added two other missing terms ("Payable to" and "Reference") and it will be available to translate once merged (and transifex translated terms are merged in Odoo on Sunday).

nle-odoo commented:
robodoo r+
